Sources/Logging/Logging.swift

```diff
@@ -803,6 +929,65 @@ extension Logger {
     public init(label: String, factory: (String) -> LogHandler) {
         self = Logger(label: label, factory(label))
     }
+
+    /// Construct a `Logger` given a `label` identifying the creator of the `Logger` and a non-standard ``Logger/MetadataProvider``.
+    ///
+    /// The `label` should identify the creator of the `Logger`. This can be an application, a sub-system, or even
+    /// a datatype.
+    ///
+    /// This initializer provides an escape hatch in case the global default logging backend implementation (set up
```

Comment:
is the escape hatch because of real world need, or more theoretical? if the latter, maybe we can add later if/when we have a real need?

Reply:
Technically this was to answer @fabianfett's request of "don't try reading task locals, there won't be any here anyway" when they'd manually set metadata in a nio handler, but looking at it that would be pretty destructive since they would also disable any user-provided even constant metadata providers -- so they likely should not be doing that... I'll remove this but wonder if it may come back someday